Symfony/CS/Tests/Fixer/Contrib/LogicalNotOperatorsWithSuccessorSpaceFixerTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tests\Fixer\Contrib;
+
+use Symfony\CS\Tests\Fixer\AbstractFixerTestBase;
+
+/**
+ * @author Javier Spagnoletti <phansys@gmail.com>
+ */
+class LogicalNotOperatorsWithSuccessorSpaceFixerTest extends AbstractFixerTestBase
+{
+    /**
+     * @dataProvider provideCases
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->makeTest($expected, $input);
+    }
+
+    public function provideCases()
+    {
+        return array(
+            array(
+                '<?php $i = 0; $i++; $foo = ! false || (! true || ! ! false);',
+                '<?php $i = 0; $i++; $foo = !false || (!true || !!false);',
+            ),
+            array(
+                '<?php $i = 0; $i--; $foo = ! false || ($i && ! true);',
+                '<?php $i = 0; $i--; $foo = !false || ($i && !true);',
+            ),
```

Comment:
ok, sorry, github short code block trim spaces, it was like:
```php
if (!    $foo) {}
```

And if multiple spaces are allowed as you said - please add a case to prove they are not modified.

-------------------

Also, what about:
```php
if (!/*foo*/$foo) {}
```
and
```php
if (!/*foo*/      $foo) {}
```
?